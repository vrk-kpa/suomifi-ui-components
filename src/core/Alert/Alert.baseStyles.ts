import { css } from 'styled-components';
import { SuomifiTheme } from '../theme';
import { element, font } from '../theme/reset';
import { transparentize } from 'polished';
import { MarginProps, buildSpacingCSS } from '../theme/utils/spacing';

export const baseStyles = (
  theme: SuomifiTheme,
  globalMargins?: MarginProps,
  propMargins?: MarginProps,
) => css`
  &.fi-alert {
    ${element(theme)}
    ${font(theme)('bodyTextSmall')}
    ${buildSpacingCSS(globalMargins)}
  ${buildSpacingCSS(propMargins, true)};
    width: 100%;

    & .fi-alert_style-wrapper {
      display: flex;
      align-items: flex-start;
      max-width: 1110px; /** Based on Suomi.fi front page header content width on large screens */
      margin: auto;

      & .fi-alert_icon {
        margin-left: ${theme.spacing.m};
        margin-top: ${theme.spacing.insetXl};
        min-height: 24px;
        min-width: 24px;
        height: 24px;
        width: 24px;
      }

      & .fi-alert_close-button {
        ${font(theme)('bodyTextSmall')}
        height: 40px;
        display: inline-block;
        padding: 7px;
        margin: 7px ${theme.spacing.xs} ${theme.spacing.insetM} 0;
        border: 1px solid transparent;
        border-radius: ${theme.radiuses.basic};
        white-space: nowrap;

        &:focus-visible {
          position: relative;
          ${theme.focuses.highContrastFocus} /* For high contrast mode */

          &:after {
            ${theme.focuses.absoluteFocus}
          }
        }
        &:active {
          background: ${theme.gradients.whiteBaseToDepthLight1};
        }
        &:hover {
          border-color: ${theme.colors.blackBase};
        }

        & .fi-icon {
          width: 14px;
          height: 14px;
          margin-left: ${theme.spacing.xxs};
          transform: translateY(0.1em);

          .fi-icon-base-fill {
            fill: currentColor;
          }
        }
      }

      & .fi-alert_text-content-wrapper {
        display: flex;
        flex-direction: column;
        padding: 0 ${theme.spacing.m};
        margin: ${theme.spacing.s} auto ${theme.spacing.s} 0;

        & .fi-alert_content {
          vertical-align: middle;
          ${font(theme)('bodyTextSmall')}
        }
      }
    }
    /** Status variant styles */
    &.fi-alert--neutral {
      background-color: ${theme.colors.infoLight1};
      border-bottom: 1px solid ${transparentize(0.8, theme.colors.infoBase)};
      & .fi-alert_style-wrapper {
        & .fi-alert_icon--neutral {
          & .fi-icon-base-fill {
            fill: ${theme.colors.blackBase};
          }
        }
      }
    }

    &.fi-alert--error {
      background-color: ${theme.colors.alertBase};
      color: ${theme.colors.whiteBase};
      & .fi-alert_style-wrapper {
        & .fi-alert_icon--error {
          & .fi-icon-base-fill {
            fill: ${theme.colors.whiteBase};
          }
        }
      }
    }

    &.fi-alert--warning {
      background-color: ${theme.colors.warningBase};
      & .fi-alert_style-wrapper {
        & .fi-alert_icon--warning {
          & .fi-icon-base-fill {
            fill: ${theme.colors.blackBase};
          }
        }
      }
    }

    /** Small screen variant styles */
    &.fi-alert--small-screen {
      & .fi-alert_style-wrapper {
        width: 100%;

        & .fi-alert_text-content-wrapper {
          padding: 0 0 0 ${theme.spacing.xs};
        }

        & .fi-alert_close-button {
          margin: 0;
          & .fi-icon {
            margin-right: ${theme.spacing.xxs};
          }
        }
      }
    }
  }
`;
