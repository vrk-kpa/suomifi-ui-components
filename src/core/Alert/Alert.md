`<Alert>` is used at the top of a web page to convey information regarding the entire page or web service. It has the ARIA role `'alert'`, which ensures the component's content is read by most screen readers immediately on page load.

If you need to show a notification regarding a more specific part of the page, use the [Notification](./#/Components/Notification) component instead.

If you need to show dynamically appearing alert, use the [InlineAlert](./#/Components/InlineAlert) component instead.

Examples:

- [Basic use](./#/Components/Alert?id=basic-use)
- [Warning status](./#/Components/Alert?id=warning-status)
- [Error status](./#/Components/Alert?id=error-status)
- [Small screen](./#/Components/Alert?id=small-screen)

<div style="margin-bottom: 40px">
  [Props & methods](./#/Components/Alert?id=props--methods)
</div>

### Basic use

```js
import { Alert, Block } from 'suomifi-ui-components';
import { useState } from 'react';

const [showAlert, setShowAlert] = useState(true);

showAlert && (
  <Block style={{ width: '900px' }}>
    <Alert
      closeText="Close"
      onCloseButtonClick={() => setShowAlert(false)}
    >
      You are using a beta version of the service. It might contain
      some bugs or glitches.
    </Alert>
  </Block>
);
```

### Warning status

The warning status of an Alert is used for conveying information which affects users but is not absolutely critical.

```js
import { Alert, Block } from 'suomifi-ui-components';
import { useState } from 'react';

const [showAlert, setShowAlert] = useState(true);

showAlert && (
  <Block style={{ width: '900px' }}>
    <Alert
      status="warning"
      closeText="Close"
      onCloseButtonClick={() => setShowAlert(false)}
    >
      The service will be temporarily unavailable on 5.6.2021 at 21.00
      â€“ 23.59 due to maintenance.
    </Alert>
  </Block>
);
```

### Error status

The error status of an Alert is used for conveying critical issues.

```js
import { Alert, Block } from 'suomifi-ui-components';
import { useState } from 'react';

const [showAlert, setShowAlert] = useState(true);

showAlert && (
  <Block style={{ width: '900px' }}>
    <Alert
      status="error"
      closeText="Close"
      onCloseButtonClick={() => setShowAlert(false)}
    >
      We are currently experiencing disruptions in the service.
      Submitting applications is not possible.
    </Alert>
  </Block>
);
```

### Small screen

Set `smallScreen` to true on narrower screens (mobile devices).

```js
import { Alert, Block } from 'suomifi-ui-components';
import { useState } from 'react';

const [showAlert, setShowAlert] = useState(true);

showAlert && (
  <Block style={{ width: '900px' }}>
    <Alert
      smallScreen
      closeText="Close"
      onCloseButtonClick={() => setShowAlert(false)}
    >
      You are using a beta version of the service. It might contain
      some bugs or glitches.
    </Alert>
  </Block>
);
```

### Props & methods

Alert component supports [margin props](./#/Spacing/Margin%20props) for spacing.
