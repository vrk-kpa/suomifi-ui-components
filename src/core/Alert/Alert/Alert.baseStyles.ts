import { css } from 'styled-components';
import { SuomifiTheme } from '../../theme';
import { font } from '../../theme/reset';
import { baseAlertBaseStyles } from '../BaseAlert/BaseAlert.baseStyles';

export const baseStyles = (theme: SuomifiTheme) => css`
  ${baseAlertBaseStyles(theme)}
  overflow: hidden;
  padding-bottom: 7px;
  & .fi-alert_icon-wrapper {
    flex: 0;
  }

  & .fi-alert_close-button {
    ${font(theme)('bodyTextSmall')}
    flex: 1 0 auto;
    flex-wrap: nowrap;
    display: flex;
    box-sizing: border-box;
    margin-top: 9px;
    margin-bottom: 0;
    max-width: 50%;
    min-width: 40px;
    text-align: right;
    padding: 7px 0 7px 8px;
    border: 1px solid transparent;
    border-radius: ${theme.radius.basic};
    text-transform: uppercase;
    flex-grow: 0;
    &:focus-visible {
      outline: 0;
      position: relative;
      &:after {
        ${theme.focus.absoluteFocus}
      }
    }
    &:active {
      background: ${theme.gradients.whiteBaseToDepthLight1};
    }
    &:hover {
      border-color: ${theme.colors.blackBase};
    }

    & svg {
      flex-grow: 1;
      flex-shrink: 0;
      padding-top: 4px;
      padding-left: 7px;
      & .fi-icon {
        margin-left: ${theme.spacing.xxs};
        margin-top: ${theme.spacing.xxs};
        display: block;
        height: 14px;
        width: 14px;
      }
    }
  }
  &:focus-visible {
    outline: 0;
    position: relative;

    &:after {
      ${theme.focus.absoluteFocus}
    }
  }
  &:active {
    background: ${theme.gradients.whiteBaseToDepthLight1};
  }
  &:hover {
    border-color: ${theme.colors.blackBase};
  }

  /** Small screen variant styles */
  &.fi-alert--small-screen {
    & .fi-alert_style-wrapper {
      padding: 0;
      margin: 0;
    }
    & .fi-alert_icon-wrapper {
      padding-left: 15px;
    }
    & .fi-alert_close-button {
      justify-content: flex-end;
      margin: 4px;
      padding: 3px 5px 7px 0;
      & .fi-icon {
        margin: 0;
      }
    }
  }
`;
