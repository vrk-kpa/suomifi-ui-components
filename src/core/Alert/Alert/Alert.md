```js
import { Alert } from 'suomifi-ui-components';

<>
  <Alert closeText="Close" smallScreen>
    The service will be temporarily unavailable on 5.6.2021 at 21.00 –
    23.59 due to maintenance.
  </Alert>
  <Alert closeText="Close Lorem ipsum dolor sit amet consectetur consectetur adipiscing elit">
    This is a short alert text
  </Alert>
  <Alert closeText="Close">
    This is a long alert text. Lorem ipsum dolor sit amet, consectetur
    adipiscing elit. Morbi est nulla, rhoncus ac odio at, blandit
    ornare mi. Suspendisse potenti. Vestibulum ante ipsum primis in
    faucibus orci luctus et ultrices posuere cubilia curae;
    Suspendisse efficitur et turpis eget tempor. Nam sed lacus ut
    lorem feugiat tincidunt. Mauris quis nisi placerat, convallis
    turpis nec, auctor massa. Nam ornare tortor id tortor tristique,
    in semper augue fermentum. Nulla at pharetra leo, a ullamcorper
    metus. Donec molestie velit tristique feugiat interdum. Donec
    ultrices efficitur pellentesque. Sed luctus ac metus sed rhoncus.
  </Alert>

  <Alert status="error" closeText="Long close text">
    Something went wrong. Please try again in a moment.
  </Alert>

  <Alert status="warning" closeText="Close">
    The service will be temporarily unavailable on 5.6.2021 at 21.00 –
    23.59 due to maintenance.
  </Alert>
</>;
```
