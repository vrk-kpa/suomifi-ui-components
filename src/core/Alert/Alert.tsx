import React, { Component, forwardRef, ReactNode } from 'react';
import { default as styled } from 'styled-components';
import classnames from 'classnames';
import {
  HtmlDiv,
  HtmlDivWithRef,
  HtmlButton,
  HtmlButtonProps,
  HtmlDivWithRefProps,
} from '../../reset';
import { AutoId } from '../utils/AutoId/AutoId';
import { getConditionalAriaProp } from '../../utils/aria';
import {
  SuomifiThemeProp,
  SuomifiThemeConsumer,
  SpacingConsumer,
} from '../theme';
import {
  separateMarginProps,
  MarginProps,
  GlobalMarginProps,
} from '../theme/utils/spacing';
import { baseStyles } from './Alert.baseStyles';
import { IconClose, IconError, IconInfo, IconWarning } from 'suomifi-icons';
import { filterDuplicateKeys } from '../../utils/common/common';

const baseClassName = 'fi-alert';
const alertClassNames = {
  styleWrapper: `${baseClassName}_style-wrapper`,
  textContentWrapper: `${baseClassName}_text-content-wrapper`,
  content: `${baseClassName}_content`,
  icon: `${baseClassName}_icon`,
  closeButton: `${baseClassName}_close-button`,
  smallScreen: `${baseClassName}--small-screen`,
};

export interface AlertProps extends HtmlDivWithRefProps, MarginProps {
  /** Style variant. Affects color and icon.
   * @default 'neutral'
   */
  status?: 'neutral' | 'warning' | 'error';
  /** Main content of the alert */
  children?: ReactNode;
  /** Use small screen styling */
  smallScreen?: boolean;
  /** Text to label the close button.
   * Text is used as `aria-label`. Text is  is visible in regular size and hidden in small screen variant.
   */
  closeText: string;
  /** Click handler for the close button */
  onCloseButtonClick?: () => void;
  /** Custom props passed to the close button */
  closeButtonProps?: Omit<HtmlButtonProps, 'onClick' | 'aria-label'>;
  /** Ref is placed to the outermost div element of the component. Alternative for React `ref` attribute. */
  forwardedRef?: React.RefObject<HTMLDivElement>;
}

class BaseAlert extends Component<AlertProps> {
  render() {
    const {
      className,
      status = 'neutral',
      children,
      id,
      closeText,
      onCloseButtonClick,
      smallScreen,
      forwardedRef,
      closeButtonProps = {},
      ...rest
    } = this.props;
    const [_marginProps, passProps] = separateMarginProps(rest);

    const {
      className: customCloseButtonClassName,
      'aria-describedby': closeButtonPropsAriaDescribedBy,
      ...closeButtonPassProps
    } = closeButtonProps;

    return (
      <HtmlDivWithRef
        as="section"
        {...passProps}
        className={classnames(baseClassName, className, {
          [`${baseClassName}--${status}`]: !!status,
          [alertClassNames.smallScreen]: !!smallScreen,
        })}
        style={{ ...passProps?.style }}
        ref={forwardedRef}
      >
        <HtmlDiv className={alertClassNames.styleWrapper}>
          {status === 'warning' && (
            <IconWarning
              className={classnames(
                alertClassNames.icon,
                `${alertClassNames.icon}--${status}`,
              )}
            />
          )}
          {status === 'error' && (
            <IconError
              className={classnames(
                alertClassNames.icon,
                `${alertClassNames.icon}--${status}`,
              )}
            />
          )}
          {status === 'neutral' && (
            <IconInfo
              className={classnames(
                alertClassNames.icon,
                `${alertClassNames.icon}--${status}`,
              )}
            />
          )}

          <HtmlDiv
            className={alertClassNames.textContentWrapper}
            id={id}
            role="alert"
          >
            <HtmlDiv className={alertClassNames.content}>{children}</HtmlDiv>
          </HtmlDiv>
          <HtmlButton
            className={classnames(
              alertClassNames.closeButton,
              customCloseButtonClassName,
            )}
            {...getConditionalAriaProp('aria-describedby', [
              id,
              closeButtonPropsAriaDescribedBy,
            ])}
            onClick={onCloseButtonClick}
            {...getConditionalAriaProp('aria-label', [closeText])}
            {...closeButtonPassProps}
          >
            {!smallScreen ? closeText : ''}
            <IconClose />
          </HtmlButton>
        </HtmlDiv>
      </HtmlDivWithRef>
    );
  }
}

const StyledAlert = styled(
  (props: AlertProps & SuomifiThemeProp & GlobalMarginProps) => {
    const { theme, globalMargins, ...passProps } = props;
    return <BaseAlert {...passProps} />;
  },
)`
  ${({ theme, globalMargins, ...rest }) => {
    const [marginProps, _passProps] = separateMarginProps(rest);
    const cleanedGlobalMargins = filterDuplicateKeys(
      globalMargins.alert,
      marginProps,
    );
    return baseStyles(theme, cleanedGlobalMargins, marginProps);
  }}
`;

const Alert = forwardRef(
  (props: AlertProps, ref: React.RefObject<HTMLDivElement>) => {
    const { id: propId, ...passProps } = props;
    return (
      <SpacingConsumer>
        {({ margins }) => (
          <SuomifiThemeConsumer>
            {({ suomifiTheme }) => (
              <AutoId id={propId}>
                {(id) => (
                  <StyledAlert
                    forwardedRef={ref}
                    theme={suomifiTheme}
                    globalMargins={margins}
                    id={id}
                    {...passProps}
                  />
                )}
              </AutoId>
            )}
          </SuomifiThemeConsumer>
        )}
      </SpacingConsumer>
    );
  },
);

Alert.displayName = 'Alert';
export { Alert };
