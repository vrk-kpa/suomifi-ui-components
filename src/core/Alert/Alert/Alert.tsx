import React, { Component, forwardRef } from 'react';
import { default as styled } from 'styled-components';
import classnames from 'classnames';
import {
  HtmlDiv,
  HtmlDivWithRef,
  HtmlButton,
  HtmlButtonProps,
} from '../../../reset';
import { Icon } from '../../../core/Icon/Icon';
import {
  BaseAlertProps,
  alertClassNames,
  baseClassName,
} from '../BaseAlert/BaseAlert';
import { AutoId } from '../../utils/AutoId/AutoId';
import { getConditionalAriaProp } from '../../../utils/aria';
import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
import { baseStyles } from './Alert.baseStyles';

export interface AlertProps extends BaseAlertProps {
  /** Text to label the close button. Visible + `aria-label` in regular size and only used as `aria-label` in small screen variant */
  closeText: string;
  /** Click handler for the close button */
  onCloseButtonClick?: () => void;
  /** Custom props passed to the close button */
  closeButtonProps?: Omit<HtmlButtonProps, 'onClick' | 'aria-label'>;
  /** Use small screen styling */
  smallScreen?: boolean;
}

interface InnerRef {
  forwardedRef: React.RefObject<HTMLDivElement>;
}

class BaseAlert extends Component<AlertProps & InnerRef> {
  render() {
    const {
      className,
      status = 'neutral',
      children,
      id,
      closeText,
      onCloseButtonClick,
      smallScreen,
      closeButtonProps = {},
      ...passProps
    } = this.props;

    const {
      className: customCloseButtonClassName,
      'aria-describedby': closeButtonPropsAriaDescribedBy,
      ...closeButtonPassProps
    } = closeButtonProps;

    const variantIcon = status === 'neutral' ? 'info' : status;
    return (
      <HtmlDivWithRef
        as="section"
        {...passProps}
        className={classnames(baseClassName, className, {
          [`${baseClassName}--${status}`]: !!status,
          [alertClassNames.smallScreen]: !!smallScreen,
        })}
      >
        <HtmlDiv className={alertClassNames.styleWrapper}>
          <HtmlDiv className={alertClassNames.iconWrapper}>
            <Icon icon={variantIcon} className={alertClassNames.icon} />
          </HtmlDiv>

          <HtmlDiv
            className={alertClassNames.textContentWrapper}
            id={id}
            role="alert"
          >
            <HtmlDiv className={alertClassNames.content}>{children}</HtmlDiv>
          </HtmlDiv>
          <HtmlButton
            className={classnames(
              alertClassNames.closeButton,
              customCloseButtonClassName,
            )}
            {...getConditionalAriaProp('aria-describedby', [
              id,
              closeButtonPropsAriaDescribedBy,
            ])}
            onClick={onCloseButtonClick}
            {...getConditionalAriaProp('aria-label', [closeText])}
            {...closeButtonPassProps}
          >
            {!smallScreen ? closeText.toUpperCase() : ''}
            <Icon icon="close" />
          </HtmlButton>
        </HtmlDiv>
      </HtmlDivWithRef>
    );
  }
}

const StyledAlert = styled(
  (props: AlertProps & InnerRef & SuomifiThemeProp) => {
    const { theme, ...passProps } = props;
    return <BaseAlert {...passProps} />;
  },
)`
  ${({ theme }) => baseStyles(theme)}
`;

export const Alert = forwardRef(
  (props: AlertProps, ref: React.RefObject<HTMLDivElement>) => {
    const { id: propId, ...passProps } = props;
    return (
      <SuomifiThemeConsumer>
        {({ suomifiTheme }) => (
          <AutoId id={propId}>
            {(id) => (
              <StyledAlert
                forwardedRef={ref}
                theme={suomifiTheme}
                id={id}
                {...passProps}
              />
            )}
          </AutoId>
        )}
      </SuomifiThemeConsumer>
    );
  },
);
