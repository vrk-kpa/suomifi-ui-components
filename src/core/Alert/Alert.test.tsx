import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Alert } from './Alert';
import { axeTest } from '../../utils/test';

describe('props', () => {
  describe('status', () => {
    const ErrorAlert = (
      <Alert status="error" closeText="Close">
        Testcontent
      </Alert>
    );

    test('error status should have correct classname applied', () => {
      const { container } = render(ErrorAlert);
      expect(container.firstChild).toHaveClass('fi-alert--error');
    });

    const WarningAlert = (
      <Alert status="warning" closeText="Close">
        General warning
      </Alert>
    );

    test('warning status should have correct classname applied', () => {
      const { container } = render(WarningAlert);
      expect(container.firstChild).toHaveClass('fi-alert--warning');
    });
  });
  describe('className', () => {
    const customClassAlert = (
      <Alert closeText="Close" className="custom-class">
        Test content
      </Alert>
    );

    it('contains base classname', () => {
      const { container } = render(customClassAlert);
      expect(container.firstChild).toHaveClass('fi-alert');
    });

    it('contains custom classname', () => {
      const { container } = render(customClassAlert);
      expect(container.firstChild).toHaveClass('custom-class');
    });
  });
  describe('children', () => {
    const alertWithElementChild = (
      <Alert closeText="Close">
        <div>Test alert</div>
      </Alert>
    );

    it('has the given content', () => {
      const { container } = render(alertWithElementChild);
      expect(container.firstChild).toHaveTextContent('Test alert');
    });

    it('should match snapshot', () => {
      const { container } = render(alertWithElementChild);
      expect(container).toMatchSnapshot();
    });
  });

  test('closeButton should have the given closeButtonProps', () => {
    const AlertWithButtonProps = (
      <Alert
        closeText="Close"
        closeButtonProps={{
          'aria-labelledby': 'test-element',
          disabled: true,
          className: 'testClass',
        }}
      >
        Testcontent
      </Alert>
    );
    const { getByRole } = render(AlertWithButtonProps);
    const closeButton = getByRole('button');

    expect(closeButton).toHaveAttribute('aria-labelledby', 'test-element');
    expect(closeButton).toHaveAttribute('disabled');
    expect(closeButton).toHaveClass('testClass');
  });
  describe('Alert role and className check', () => {
    const DefaultAlert = (
      <Alert id="testIdDefault" closeText="Close">
        Testcontent
      </Alert>
    );

    it('should have alert role by default', () => {
      const { container } = render(DefaultAlert);
      expect(container.querySelector('#testIdDefault')).toHaveClass(
        'fi-alert_text-content-wrapper',
      );
      expect(container.querySelector('#testIdDefault')).toHaveAttribute(
        'role',
        'alert',
      );
    });
  });

  describe('margin prop', () => {
    it('should have margin style from margin prop', () => {
      const { container } = render(<Alert closeText="Close" margin="xs" />);
      expect(container.firstChild).toHaveStyle('margin: 10px');
    });

    it('should have margin style overwritten from style', () => {
      const { container } = render(
        <Alert closeText="Close" margin="xs" style={{ margin: 2 }} />,
      );
      expect(container.firstChild).toHaveAttribute('style', 'margin: 2px;');
    });
  });

  test('onClick event is called when clicked', () => {
    const mockClick = jest.fn();
    const { getByRole } = render(
      <Alert closeText="Close" onClick={mockClick}>
        Test content
      </Alert>,
    );
    const button = getByRole('button');
    fireEvent.click(button);
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
});

describe('accessibility', () => {
  const TestAlert = <Alert closeText="Close">Testcontent</Alert>;
  test('should not have basic accessibility issues', axeTest(TestAlert));
});
