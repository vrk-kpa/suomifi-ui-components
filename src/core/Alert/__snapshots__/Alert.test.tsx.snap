// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props children should match snapshot 1`] = `
.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c4:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c4::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c4::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c4::-webkit-inner-spin-button {
  height: auto;
}

.c4::-webkit-outer-spin-button {
  height: auto;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c5 {
  vertical-align: baseline;
}

.c5.fi-icon {
  display: inline-block;
}

.c5 .fi-icon-base-fill {
  fill: currentColor;
}

.c5 .fi-icon-base-stroke {
  stroke: currentColor;
}

.c5.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c5.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c3 {
  vertical-align: baseline;
}

.c3.fi-icon {
  display: inline-block;
}

.c3 .fi-icon-base-fill {
  fill: currentColor;
}

.c3 .fi-icon-base-stroke {
  stroke: currentColor;
}

.c3.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c3.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c1.fi-alert {
  color: hsl(0,0%,13%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  width: 100%;
}

.c1.fi-alert .fi-alert_style-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  max-width: 1110px;
  margin: auto;
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_icon {
  margin-left: 20px;
  margin-top: 16px;
  min-height: 24px;
  min-width: 24px;
  height: 24px;
  width: 24px;
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_close-button {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  height: 40px;
  display: inline-block;
  padding: 7px;
  margin: 7px 10px 8px 0;
  border: 1px solid transparent;
  border-radius: 2px;
  white-space: nowrap;
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_close-button:focus-visible {
  position: relative;
  outline: 3px solid transparent;
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_close-button:focus-visible:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_close-button:active {
  background: linear-gradient(-180deg,hsl(202,7%,80%) 0%,hsla(0,0%,100%,0) 100%);
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_close-button:hover {
  border-color: hsl(0,0%,13%);
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_close-button .fi-icon {
  width: 14px;
  height: 14px;
  margin-left: 5px;
  -webkit-transform: translateY(0.1em);
  -ms-transform: translateY(0.1em);
  transform: translateY(0.1em);
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_close-button .fi-icon .fi-icon-base-fill {
  fill: currentColor;
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_text-content-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0 20px;
  margin: 15px auto 15px 0;
}

.c1.fi-alert .fi-alert_style-wrapper .fi-alert_text-content-wrapper .fi-alert_content {
  vertical-align: middle;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c1.fi-alert.fi-alert--neutral {
  background-color: hsl(216,100%,96%);
  border-bottom: 1px solid rgba(23,112,238,0.2);
}

.c1.fi-alert.fi-alert--neutral .fi-alert_style-wrapper .fi-alert_icon--neutral .fi-icon-base-fill {
  fill: hsl(0,0%,13%);
}

.c1.fi-alert.fi-alert--error {
  background-color: hsl(3,59%,43%);
  color: hsl(0,0%,100%);
}

.c1.fi-alert.fi-alert--error .fi-alert_style-wrapper .fi-alert_icon--error .fi-icon-base-fill {
  fill: hsl(0,0%,100%);
}

.c1.fi-alert.fi-alert--warning {
  background-color: hsl(42,93%,50%);
}

.c1.fi-alert.fi-alert--warning .fi-alert_style-wrapper .fi-alert_icon--warning .fi-icon-base-fill {
  fill: hsl(0,0%,13%);
}

.c1.fi-alert.fi-alert--small-screen .fi-alert_style-wrapper {
  width: 100%;
}

.c1.fi-alert.fi-alert--small-screen .fi-alert_style-wrapper .fi-alert_text-content-wrapper {
  padding: 0 0 0 10px;
}

.c1.fi-alert.fi-alert--small-screen .fi-alert_style-wrapper .fi-alert_close-button {
  margin: 0;
}

.c1.fi-alert.fi-alert--small-screen .fi-alert_style-wrapper .fi-alert_close-button .fi-icon {
  margin-right: 5px;
}

<div>
  <section
    class="c0 fi-alert c1 fi-alert--neutral"
  >
    <div
      class="c2 fi-alert_style-wrapper"
    >
      <svg
        aria-hidden="true"
        class="fi-icon c3 fi-alert_icon fi-alert_icon--neutral"
        focusable="false"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="fi-icon-base-fill"
          d="M12 0c6.617 0 12 5.383 12 12s-5.383 12-12 12S0 18.617 0 12 5.383 0 12 0Zm0 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2Zm.003 8a1 1 0 0 1 1 1v6h1a1 1 0 0 1 0 2H10a1 1 0 0 1 0-2h1v-5h-1a1 1 0 0 1 0-2h2.003Zm-.501-5a1.5 1.5 0 0 1 .145 2.994L11.502 8H11.5a1.5 1.5 0 1 1 0-3h.002Z"
          fill="#222"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="c2 fi-alert_text-content-wrapper"
        id="7"
        role="alert"
      >
        <div
          class="c2 fi-alert_content"
        >
          <div>
            Test alert
          </div>
        </div>
      </div>
      <button
        aria-describedby="7"
        aria-label="Close"
        class="c4 fi-alert_close-button"
        type="button"
      >
        Close
        <svg
          aria-hidden="true"
          class="fi-icon c5"
          focusable="false"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="fi-icon-base-fill"
            d="M4.75 3.3 12 10.55l7.25-7.25a1.026 1.026 0 0 1 1.45 1.45L13.45 12l7.25 7.25a1.026 1.026 0 0 1-1.45 1.45L12 13.45 4.75 20.7a1.026 1.026 0 0 1-1.45-1.45L10.55 12 3.3 4.75A1.026 1.026 0 0 1 4.75 3.3Z"
            fill="#222"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </section>
</div>
`;
